### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by barmac**
This PR adds TypeScript definition for better developer experience (suggestions in the editor).

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for the PR, but I'm not interested in JSDoc. However, I'm happy to accept TypeScript types if you follow [this styleguide](https://github.com/sindresorhus/typescript-definition-style-guide)?

Some examples:
- https://github.com/sindresorhus/semver-regex/commit/039944bb5b3876913171b895488874b5c200e458
- https://github.com/sindresorhus/global-dirs/commit/0aae4d64887f77af94c4fb7ca12aa9f93c0a8291
- https://github.com/sindresorhus/md5-hex/commit/bbbfeff840f26973fbffc16d4863ae6d120079f3

**Comment by barmac** [comment]
Sure, I can do that! Please have a look now :)

File: package.json [hunk 1]
```diff
@@ -44,9 +45,12 @@
 		"find-up": "^4.1.0"
 	},
 	"devDependencies": {
+		"@types/node": "^14.14.20",
 		"electron": "^8.0.0",
+		"tsd": "^0.14.0",
 		"xo": "^0.26.1"
 	},
+	"types": "index.d.ts",
```
Comment (sindresorhus) [inline package.json:53]: This is not needed.

File: index.test-d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {expectType} from 'tsd';
+import setupElectronReloader = require('.');
+
+expectType<void>(setupElectronReloader(module));
+expectType<void>(setupElectronReloader(module, { watchRenderer: true }));
```
Comment (sindresorhus) [inline index.test-d.ts:5]:
```suggestion
expectType<void>(setupElectronReloader(module, {watchRenderer: true}));
```

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/// <reference types="node"/>
+
+/**
+ * Setup `electron-reloader`.
+ *
+ * @param moduleObject - Global `module` object.
+ * @param options.watchRenderer - Watch files used in the renderer process and reload the window when they change. Defaults to `true`.
+ * @param options.debug - Prints watched paths and when files change. Can be useful to make sure you set it up correctly.
+ * @param options.ignore - Ignore patterns passed to [`chokidar`](https://github.com/paulmillr/chokidar#path-filtering). By default, files/directories starting with a `.`, `.map` files, and `node_modules` directories are ignored. This option is additive to those.
+ *
```
Comment (sindresorhus) [inline index.d.ts:8]: Drop the `*`.

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * try {
+ * 	setupElectronReloader(module, { ignore: [ 'client/src' ] });
+ * } catch (error) {
+ * 	console.error('Unable to setup reloading', error);
+ * }
+ * ```
+ */
+declare function setupElectronReloader(
+	moduleObject: NodeModule,
+	options?: {
```
Comment (sindresorhus) [inline index.d.ts:22]: Options should be a separate type and each option should have doc comments copied from the readme. See the guidelines.

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * @example
+ * ```
+ * const setupElectronReloader = require('electron-reloader');
+ *
+ * try {
+ * 	setupElectronReloader(module, { ignore: [ 'client/src' ] });
+ * } catch (error) {
+ * 	console.error('Unable to setup reloading', error);
+ * }
```
Comment (sindresorhus) [inline index.d.ts:17]: Use the example from the readme.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by barmac** [comment]
I've implemented the suggestions from your comments and think that it's ready for the second round 👍

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**